src/test/java/org/pesmypetcare/webservice/service/KcalAverageServiceTest.java

```diff
@@ -0,0 +1,132 @@
+    }
+
+    @Test
+    public void shouldReturnNothingWhenKcalAverageCreated() {
+        service.createKcalAverage(owner, petName, date, kcalAverageAverageEntity);
+        verify(kcalAverageDao).createKcalAverage(isA(String.class), isA(String.class), isA(String.class),
+            isA(KcalAverageEntity.class));
+    }
+
+    @Test
+    public void shouldReturnNothingWhenKcalAverageDeleted() throws DatabaseAccessException {
+        service.deleteKcalAverageByDate(owner, petName, date);
+        verify(kcalAverageDao).deleteKcalAverageByDate(isA(String.class), isA(String.class), isA(String.class));
+    }
+
+    @Test
+    public void shouldReturnNothingWhenAllKcalAveragesDeleted() throws DatabaseAccessException {
+        service.deleteAllKcalAverages(owner, petName);
+        verify(kcalAverageDao).deleteAllKcalAverages(isA(String.class), isA(String.class));
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenAllKcalAverageDeleteFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(kcalAverageDao).deleteAllKcalAverages(any(String.class),
+                isA(String.class));
+            service.deleteAllKcalAverages(owner, petName);
+        }, "Should return a database access exception when a kcalAverageAverage deletion fails");
+    }
+
+    @Test
+    public void shouldReturnMealEntityWhenKcalAverageRetrieved() throws DatabaseAccessException {
+        when(kcalAverageDao.getKcalAverageByDate(owner, petName, date)).thenReturn(kcalAverageAverageEntity);
+        KcalAverageEntity meal = service.getKcalAverageByDate(owner, petName, date);
+        assertSame(kcalAverageAverageEntity, meal, "Should return a kcalAverageAverage entity");
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenGetKcalAverageRequestFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(kcalAverageDao).getKcalAverageByDate(any(String.class), any(String.class),
+                isA(String.class));
+            service.getKcalAverageByDate(owner, petName, date);
+        }, "Should return an exception when retrieving a kcalAverageAverage fails");
+    }
+
+    @Test
+    public void shouldReturnMealEntityListWhenGetSetOfKcalAveragesRetrieved() throws DatabaseAccessException {
+        when(kcalAverageDao.getAllKcalAverage(owner, petName)).thenReturn(kcalAverageAverageList);
+        List<Map<String, Object>> list = service.getAllKcalAverage(owner, petName);
+        assertSame(kcalAverageAverageList, list, "Should return a list of kcalAverageAverages entities");
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenGetSetOfKcalAveragesRequestFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(kcalAverageDao).getAllKcalAverage(any(String.class), isA(String.class));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 128).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5400494)